### PR TITLE
Unit tests for LwM2M Client Utitilies

### PR DESCRIPTION
**Pull request by SeppoTakalo**
This contains also #8943 

Uses FFF for unittesting, so we could use same framework on upstream side.

### CODE REVIEW

File: tests/subsys/net/lib/lwm2m_client_utils/testcase.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+tests:
+  subsys.net.lib.lwm2m_client_utils.unittest:
+    platform_allow: native_posix
```
Comment (gopiotr) [inline tests/subsys/net/lib/lwm2m_client_utils/testcase.yaml:3]: Please add `integration_platforms` option here, to make it possible to run those tests in CI
```yaml
integration_platforms:
  - native_posix
```

**Review by gopiotr — COMMENTED** [review]
Please add register into CODEOWNERS file (https://github.com/nrfconnect/sdk-nrf/blob/main/CODEOWNERS) with path to test `/tests/subsys/net/lib/lwm2m_client_utils/` and person/people responsible for this test.

File: tests/subsys/net/lib/lwm2m_client_utils/testcase.yaml (hunk 1 above)
Comment (SeppoTakalo) [reply tests/subsys/net/lib/lwm2m_client_utils/testcase.yaml]: added

**Comment by SeppoTakalo** [comment]
> Please add register into CODEOWNERS file (https://github.com/nrfconnect/sdk-nrf/blob/main/CODEOWNERS) with path to test `/tests/subsys/net/lib/lwm2m_client_utils/` and person/people responsible for this test.

Thanks. Added.

**Review by VeijoPesonen — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by jheiskan81 — APPROVED** [review]

**Pull request merged**